frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -206,7 +210,28 @@ export const NavBar = withRouter<NavBarProps>(({ pages, baseURL, basePath }) =>
     </>
   );
 
-  return <ul className="co-m-horizontal-nav__menu">{tabs}</ul>;
+  const activePage = pages.find(({ href, path }) => {
+    const matchURL = matchPath(location.pathname, {
+      path: `${basePath}/${removeLeadingSlash(path || href)}`,
+      exact: true,
+    });
+    return matchURL?.isExact;
+  });
+
+  const labelId =
+    (activePage?.nameKey && activePage?.nameKey.split('~')[1]) || activePage?.name || 'Details';
+  return (
+    <>
+      {titlePrefix && (
+        <Helmet>
+          <title data-telemetry={`${telemetryPrefix} · ${labelId}`}>
```

Comment:
In theory telemetryPrefix can be undefined (or empty), let us unsure that we don't pass it then:

```suggestion
          <title data-telemetry={telemetryPrefix ? `${telemetryPrefix} · ${labelId}` : labelId}>
```